crates/sdk/src/api/method/live.rs

```diff
@@ -41,110 +40,99 @@ where
 			Resource::Range(range) => {
-				let record = range.into_inner();
-
-				let val::RecordIdKey::Range(range) = record.key else {
-					panic!("invalid resource?");
+				let record = range.0;
+				let RecordIdKey::Range(ref key_range) = record.key else {
+					return Err(Error::InvalidParams("Invalid range in resource".to_owned()).into());
 				};
 
-				stmt.what = Expr::Table(record.table.clone());
-
-				let id = Expr::Idiom(Idiom::field("id".to_string()));
-
-				let left = match range.start {
-					std::ops::Bound::Included(x) => Some(Expr::Binary {
-						left: Box::new(id.clone()),
-						op: BinaryOperator::MoreThanEqual,
-						right: Box::new(Expr::Literal(Literal::RecordId(
-							crate::core::expr::RecordIdLit {
-								table: record.table.clone(),
-								key: x.into_literal(),
-							},
-						))),
-					}),
-					std::ops::Bound::Excluded(x) => Some(Expr::Binary {
-						left: Box::new(id.clone()),
-						op: BinaryOperator::MoreThan,
-						right: Box::new(Expr::Literal(Literal::RecordId(
-							crate::core::expr::RecordIdLit {
-								table: record.table.clone(),
-								key: x.into_literal(),
-							},
-						))),
-					}),
-					std::ops::Bound::Unbounded => None,
-				};
-				let right = match range.end {
-					std::ops::Bound::Included(x) => Some(Expr::Binary {
-						left: Box::new(id),
-						op: BinaryOperator::LessThanEqual,
-						right: Box::new(Expr::Literal(Literal::RecordId(
-							crate::core::expr::RecordIdLit {
-								table: record.table,
-								key: x.into_literal(),
-							},
-						))),
-					}),
-					std::ops::Bound::Excluded(x) => Some(Expr::Binary {
-						left: Box::new(id),
-						op: BinaryOperator::LessThan,
-						right: Box::new(Expr::Literal(Literal::RecordId(
-							crate::core::expr::RecordIdLit {
-								table: record.table,
-								key: x.into_literal(),
-							},
-						))),
-					}),
-					std::ops::Bound::Unbounded => None,
-				};
+				// Build WHERE clause for range queries
+				let mut conditions = Vec::new();
+
+				// Handle start bound
+				match &key_range.start {
+					std::ops::Bound::Included(key) => {
+						conditions.push(format!("id >= {}:{}", record.table, key.to_sql()));
+					}
+					std::ops::Bound::Excluded(key) => {
+						conditions.push(format!("id > {}:{}", record.table, key.to_sql()));
+					}
+					std::ops::Bound::Unbounded => {}
+				}
 
-				let cond = match (left, right) {
-					(Some(l), Some(r)) => Some(Cond(Expr::Binary {
-						left: Box::new(l),
-						op: BinaryOperator::And,
-						right: Box::new(r),
-					})),
-					(Some(x), None) | (None, Some(x)) => Some(Cond(x)),
-					_ => None,
-				};
+				// Handle end bound
+				match &key_range.end {
+					std::ops::Bound::Included(key) => {
+						conditions.push(format!("id <= {}:{}", record.table, key.to_sql()));
+					}
+					std::ops::Bound::Excluded(key) => {
+						conditions.push(format!("id < {}:{}", record.table, key.to_sql()));
```

Comment:
Same escape problem here, you can just always escape the table name here with ``` ` ``` but then you need to remove any ``` ` ``` that is within the table name.